### PR TITLE
Fix catkin lint

### PR DESCRIPTION
**Pull request by Olavhaasie**
Closes PM-56

## Description
This PR introduces fixes to the `CMakeLists.txt` and `package.xml` based on warnings from `catkin_lint`.

## Changes
* Fixed package descriptions to not be generic
* Sorted package lists
* Make `CMakeLists.txt` more consistent (indentation, naming)
* Fix python packages to include a script to launch them
* Remove unnecessary `__init__.py` files
* Add install targets for shared resources, binaries and libraries (necessary for colcon)
* Fix package dependencies

**There are a lot of changes here, but the most important thing is that everything still builds and tests still succeed, so try `catkin build` locally**

### CODE REVIEW

File: march_gait_scheduler/launch/gait_scheduler.launch [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <launch>
-    <node name="gait_scheduler_node" pkg="march_gait_scheduler" type="gait_scheduler_node" output="screen"/>
+    <node name="gait_scheduler_node" pkg="march_gait_scheduler" type="march_gait_scheduler_node" output="screen"/>
```
Comment (Roelemans) [inline march_gait_scheduler/launch/gait_scheduler.launch]: Moet de name niet ook march_gait_scheduler zijn?

File: march_gait_scheduler/test/main-test.test [hunk 2]
```diff
@@ -1,4 +1,4 @@
 <launch>
-    <node name="gait_scheduler_node" pkg="march_gait_scheduler" type="gait_scheduler_node" output="screen"/>
+    <node name="gait_scheduler_node" pkg="march_gait_scheduler" type="march_gait_scheduler_node" output="screen"/>
```
Comment (Roelemans) [inline march_gait_scheduler/test/main-test.test]: Hier weer

File: march_launch/launch/checks/git_branch.launch [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <launch>
     <machine name="march" address="march" user="march" env-loader="/home/march/Documents/march_ws/devel/env.sh"/>
 
-    <node name="git_branch_check" pkg="march_launch" type="git_branch_check.py" machine="march"/>
-</launch>
\ No newline at end of file
+    <node name="git_branch_check" pkg="march_launch" type="git_branch_check" machine="march"/>
```
Comment (Roelemans) [inline march_launch/launch/checks/git_branch.launch]: Moet hier de type niet ook git_branch_check_node zijn zoals bij de anderen?

File: march_rqt_gait_selection/launch/march_rqt_gait_selection.launch [hunk 4]
```diff
@@ -1,3 +1,3 @@
 <launch>
-    <node name="march_rqt_gait_selection" pkg="march_rqt_gait_selection" type="rqt_gait_selection" output="screen"/>
+    <node name="march_rqt_gait_selection" pkg="march_rqt_gait_selection" type="march_rqt_gait_selection" output="screen"/>
```
Comment (Roelemans) [inline march_rqt_gait_selection/launch/march_rqt_gait_selection.launch]: March_rqt_gait_selection_node

Comment (Olavhaasie) [reply march_rqt_gait_selection/launch/march_rqt_gait_selection.launch]: Rqt programs are not followed by `_node`

**Pull request merged**